packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1124,10 +1148,12 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
     _updateInvertColors();
     _resolveImage();
 
-    if (TickerMode.of(context)) {
-      _listenToStream();
-    } else {
+    _isPaused = !TickerMode.of(context) || (MediaQuery.maybeDisableAnimationsOf(context) ?? false);
+
+    if (_isPaused && _frameNumber != null) {
```

Comment:
do you still need to keep track of number of frames? or can this be a boolean field?

Reply:
I was about to change it but I realized it is used to give the frame number to frameBuilder: https://main-api.flutter.dev/flutter/widgets/ImageFrameBuilder.html